applications/ParticleMechanicsApplication/custom_constitutive/hencky_mc_plane_strain_2D_law.cpp

```diff
@@ -0,0 +1,75 @@
+//    |  /           | 
+//    ' /   __| _` | __|  _ \   __| 
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/ 
+//                   Multi-Physics  
+//
+//  License:		BSD License 
+//					Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ilaria Iaconeta
+//
+
+// System includes
+#include <iostream>
+
+// External includes
+#include<cmath>
+
+// Project includes
+#include "includes/properties.h"
+#include "custom_constitutive/hencky_mc_plane_strain_2D_law.hpp"
+#include "particle_mechanics_application.h"
+
+namespace Kratos
+{
+
+//******************************CONSTRUCTOR*******************************************
+//************************************************************************************
+
+HenckyMCPlasticPlaneStrain2DLaw::HenckyMCPlasticPlaneStrain2DLaw()
+    : HenckyElasticPlasticPlaneStrain2DLaw()
+{
+  mpHardeningLaw   = HardeningLaw::Pointer( new HardeningLaw() );
```

Comment:
is there a good reason for using a pointer here?

wouldn't it be better to just define a class member? note that this would be a pretty significative memory saving